spec/Functions.tex

```diff
@@ -593,16 +593,27 @@ \subsubsection{The Default Intent}
 \chpl{sync}    & \chpl{ref}   \\
 \chpl{single}  & \chpl{ref}   \\
 \chpl{atomic}  & \chpl{ref}   \\
-\chpl{record}  & \chpl{const} \\
+\chpl{record}  & \chpl{const} (but see below) \\
 \chpl{class}   & \chpl{const} \\
 \chpl{union}   & \chpl{const} \\
 \chpl{dmap}    & \chpl{const} \\
 \chpl{domain}  & \chpl{const} \\
-array          & \chpl{ref}   \\
+array          & \chpl{ref} / \chpl{const ref} \\
 \hline
 \end{tabular}
 \end{center}
 
+The default intent for arrays is \chpl{ref} or \chpl{const ref} - meaning
+that the array is not copied by default. The compiler will choose
+\chpl{ref} or \chpl{const ref} based upon whether or not the formal
+argument is modified inside of the function. The choice between
+\chpl{ref} and \chpl{const ref} is similar to and interacts with the
+ref-pair feature (see \rsec{Ref_Pair}).
+
+The default intent for the implicit \chpl{this} argument for records is
+\chpl{ref} or \chpl{const ref} depending on whether the formal argument
+is modified inside of the function.
```

Comment:
While gathering links for the CHANGES file, I'm noting that this seems like it ended up in the wrong place and feels like somewhat of a non-sequitur here.  This section is about default intents for formal arguments.  While 'this' is a formal argument in the implementation, for users, the spec section dealing with how 'this' is passed is in Classes.tex "The Method Receiver and the this Argument" (search on 'this-intent')